### DISCUSSION TITLE
where is counterReducer from?

### DISCUSSION THREAD

**Discussion by snowden-fu** [reduxjs/redux#4403] [Help] [answered]
In [official demo](https://redux.js.org/tutorials/essentials/part-2-app-structure#creating-slice-reducers-and-actions), I have noticed `counterReducer` is from `counterSlice`, but actually, there is no `counterReducer` in `counterSlice` file, so i am wondering where is `counterReducer` from?
``` javascript
import { configureStore } from '@reduxjs/toolkit'
import counterReducer from '../features/counter/counterSlice'

export default configureStore({
  reducer: {
    counter: counterReducer
  }
})
```

below is `counterSlice.js`
``` javascript
import { createSlice } from '@reduxjs/toolkit'

export const counterSlice = createSlice({
  name: 'counter',
  initialState: {
    value: 0
  },
  reducers: {
    increment: state => {
      // Redux Toolkit allows us to write "mutating" logic in reducers. It
      // doesn't actually mutate the state because it uses the immer library,
      // which detects changes to a "draft state" and produces a brand new
      // immutable state based off those changes
      state.value += 1
    },
    decrement: state => {
      state.value -= 1
    },
    incrementByAmount: (state, action) => {
      state.value += action.payload
    }
  }
})

export const { increment, decrement, incrementByAmount } = counterSlice.actions

export default counterSlice.reducer
```

**Comment by markerikson** [comment, marked as answer]
It's an ES module "default import/export".

When you do `export default whatever`, another file can do `import AnyNameYouWantHere from "./some/other/file"`.

In this case, I called it `counterReducer`, but I could have called it `import fred` or `import aldkfjalsdkflkasjdf`.  It's just creating a variable name in the importing file.
